owncloudApp/src/main/java/com/owncloud/android/sharees/presentation/SearchShareesFragment.kt

```diff
@@ -0,0 +1,224 @@
+            searchManager.getSearchableInfo(
+                activity!!.componentName
+            )   // assumes parent activity is the searchable activity
+        )
+        searchView.setIconifiedByDefault(false)    // do not iconify the widget; expand it by default
+
+        searchView.imeOptions = EditorInfo.IME_FLAG_NO_EXTRACT_UI // avoid fullscreen with softkeyboard
+
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query: String): Boolean {
+                Log_OC.v(TAG, "onQueryTextSubmit intercepted, query: $query")
+                return true    // return true to prevent the query is processed to be queried;
+                // a user / group will be picked only if selected in the list of suggestions
+            }
+
+            override fun onQueryTextChange(newText: String): Boolean {
+                return false   // let it for the parent listener in the hierarchy / default behaviour
+            }
+        })
+
+        return view
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        activity!!.setTitle(R.string.share_with_title)
+
+        // Load private shares in the list
+        listener?.refreshPrivateShares()
+    }
+
+    fun updatePrivateShares(privateShares: ArrayList<OCShare>) {
+        // Update list of users/groups
+        userGroupsAdapter = ShareUserListAdapter(
+            activity!!.applicationContext,
+            R.layout.share_user_item, privateShares, this
+        )
+
+        // Show data
+        val usersList = view!!.findViewById<ListView>(R.id.searchUsersListView)
+
+        if (privateShares.size > 0) {
+            usersList.visibility = View.VISIBLE
+            usersList.adapter = userGroupsAdapter
+
+        } else {
+            usersList.visibility = View.GONE
+        }
+    }
+
+    override fun onAttach(activity: Activity?) {
+        super.onAttach(activity)
+        try {
+            listener = activity as ShareFragmentListener?
+        } catch (e: ClassCastException) {
+            throw ClassCastException(activity!!.toString() + " must implement OnFragmentInteractionListener")
+        }
+
+    }
+
+    override fun onStart() {
+        super.onStart()
+        // focus the search view and request the software keyboard be shown
+        val searchView = view!!.findViewById<View>(R.id.searchView)
+        if (searchView.requestFocus()) {
+            val imm = activity!!.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+            imm?.showSoftInput(searchView.findFocus(), InputMethodManager.SHOW_IMPLICIT)
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        hideSoftKeyboard()
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        listener = null
+    }
+
+    private fun hideSoftKeyboard() {
+        if (view != null) {
```

Comment:
do it kotlin

            view?.findViewById<View>(R.id.searchView)?.let {
                val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
                imm?.hideSoftInputFromWindow(searchView.windowToken, 0)
            }

Reply:
Done